library/net.c

```diff
@@ -1217,7 +1217,7 @@ struct dnet_net_state *dnet_state_create(struct dnet_node *n,
 err_out_dup_destroy:
 	dnet_sock_close(n, st->write_s);
 err_out_free:
-	free(st);
+	dnet_state_destroy(st);
```

Comment:
`dnet_state_destroy` calls `pthread_mutex_lock(&st->trans_lock)` as example. It'll be work correctly after destroy of mutex?

Reply:
Yes, you're right. I will add WIP label and rework it